File: src/main/java/net/hectus/neobb/turn/default_game/warp/TWoodWarp.java
package net.hectus.neobb.turn.default_game.warp;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.turn.default_game.attributes.clazz.Clazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.NatureClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import org.bukkit.World;

import java.util.List;

public class TWoodWarp extends WarpTurn {
    public TWoodWarp(World world) { super(world, "wood"); }
    public TWoodWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "wood"); }

    @Override
    public int chance() {
        return 100;
    }

    @Override
    public List<Class<? extends Clazz>> allows() {
        return List.of(NeutralClazz.class, NatureClazz.class);
    }

    @Override
    public int cost() {
        return 4;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/warp/TWoodWarp.kt
package net.hectus.neobb.turn.default_game.warp

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.structure.StaticStructure
import net.hectus.neobb.structure.StaticStructures
import net.hectus.neobb.turn.default_game.attribute.clazz.Clazz
import net.hectus.neobb.turn.default_game.attribute.clazz.NatureClazz
import net.hectus.neobb.turn.default_game.attribute.clazz.NeutralClazz
import kotlin.reflect.KClass

class TWoodWarp(data: PlacedStructure?, cord: Cord, player: NeoPlayer?
                ) : WarpTurn(data, cord, "wood", player) {
    override val staticStructure: StaticStructure = StaticStructures.Default.Warp.WOOD

    override val cost: Int = 4
    override val chance: Double = 100.0
    override val allows: List<KClass<out Clazz>> = listOf(NeutralClazz::class, NatureClazz::class)
    override val temperature: Temperature = Temperature.NORMAL
}
